### ISSUE TITLE
Cannot build engines with multiple cores (20240318 and 20240319)

### ISSUE THREAD

**Issue by schrummy14** [OpenRadioss/OpenRadioss#2428]
**Describe the bug**
Engine files fail to build when using more than one thread.
Version 20240315 built without issue.
Versions 20240318 and 20240319 both show the build errors.
Happens with both the smp/mpp and dp/sp versions of the engines.

**To Reproduce**
Tested on Rocky Linux 9 with the following packages:
```bash
dnf update -y &&\
dnf group install -y "Development Tools" &&\
dnf install -y \
    gcc gcc-gfortran gcc-c++ make cmake perl git-lfs \
    wget git patch diffutils libxcrypt-compat \
    which python openmpi-devel
```

Download latest release (20240319) and run the command:
```bash
cd /opt/OpenRadioss/engine
./build_script.sh -arch=linux64_gf -nt=32 -prec=dp -debug=0 -release
```

**Expected behavior**
Engine files build

**Output**
```bash
/opt/OpenRadioss/engine/source/output/anim/generate/anim_nodal_vector_fvmbags.F:45:10:

   45 |       USEGROUPDEF_MOD , only:GROUP_
      |          1
Fatal Error: Cannot open module file ‘groupdef_mod.mod’ for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [CMakeFiles/engine_linux64_gf.dir/build.make:35862: CMakeFiles/engine_linux64_gf.dir/source/output/anim/generate/anim_nodal_vector_fvmbags.F.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/engine_linux64_gf.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Desktop (please complete the following information):**
 - Version tag: 20240319
- Hardware: AMD Ryzen 9 5950X 16-Core Processor
- Configuration: Failed on 2, 4, 8, 16, and 32 threads

**Additional context**
Is able to be built after the failure. It seems like the mod file is not being reported as a dependency.

**Comment by schrummy14** [comment]
It does look like the file that the error is happening on was updated for the 20240318 release:
https://github.com/OpenRadioss/OpenRadioss/commit/92a77276bf7f9cd9696a0112993de311f13fb0ae

**Comment by OlivierWienholtz** [comment]
Hello Matt,

Thank you for pointing this out.
It is indeed a Dependency issue where Cmake did not find the dependency to GROUPDEF_MOD.
The reason is that USE & Modulename were attached. It is correct in Fortran syntax, but cause dependency tools not to recognize the Module & applies the dependency. The consequence is that this file is compiled before the Module and this error appear.

I usually build up to 16 cores and did not bump in :)

**Comment by OlivierWienholtz** [comment]
The fix is done with Commit  : 921b4ffb551bf6173d3a0a5b81b52826fc14a5b1

**Comment by schrummy14** [comment]
Hello,
Thank you for the reply. I did find some additional locations where there was a space missing.
I have created a pull request:
#2434
